# HG changeset patch
# User Rik <rik@octave.org>
# Date 1346042488 25200
#      Sun Aug 26 21:41:28 2012 -0700
# Node ID a23693fa5ba9739a7b881441bdd244636fd6114a
# Parent  a38d5ed5b3d894aa1812250b926f1490afd790cb
Add mwSignedIndex type for better mex compatibility with Matlab (bug #37133)

mxarray.in.h: Add mwSignedIndex type which is a typedef for OCTAVE_IDX_TYPE.

diff --git a/libinterp/interp-core/mxarray.in.h b/libinterp/interp-core/mxarray.in.h
--- a/libinterp/interp-core/mxarray.in.h
+++ b/libinterp/interp-core/mxarray.in.h
@@ -83,16 +83,17 @@ typedef char mxChar;
 
 /*
  * FIXME? Mathworks says these should be size_t on 64-bit system and when
  * mex is used with the -largearraydims flag, but why do that? Its better
  * to conform to the same indexing as the rest of Octave
  */
 typedef %OCTAVE_IDX_TYPE% mwSize;
 typedef %OCTAVE_IDX_TYPE% mwIndex;
+typedef %OCTAVE_IDX_TYPE% mwSignedIndex;
 
 #if ! defined (MXARRAY_TYPEDEFS_ONLY)
 
 #include <cstring>
 
 class octave_value;
 
 #define DO_MUTABLE_METHOD(RET_T, METHOD_CALL) \
